### PR TITLE
Added wait_for_confirmation() to AlgodClient

### PR DESCRIPTION
**Pull request by DamianB-BitFlipper**
When using the Python SDK, I oftentimes find myself copying the same code to wait_for_confirmation() of a pending transaction.

The wait_for_confirmation() waits for a pending transaction to be accepted and confirmed by the network. Until then, it blocks. I have implemented this function with a timeout feature as well that will raise an Exception if the timeout is achieved. 

I also went through the code and found spots where a rigged "wait_for_confirmation" like code was used to wait for the pending transaction to be accepted. Instead, I replaced those instances with the proper wait_for_confirmation()

I am unsure what to do about the CHANGELOG

### CODE REVIEW

**Review by cremeikas — APPROVED** [review]
Was looking for almost exactly this functionality, and came across the PR by accident.

Seems to be working great in my test cases.

File: algosdk/future/transaction.py
Comment (algorandskiy) [inline algosdk/future/transaction.py:3012]:
```suggestion
def wait_for_confirmation(algod_client, txid, wait_rounds=1, **kwargs):
```
Rationale:
txid is most used ident for txn ids across all the code
timeout usually assumes time, but the func accepts rounds

default value needs to be meaningful since the func is supposed to wait at least round.
So I suggest to set `current_round = last_round + wait_rounds` below

File: algosdk/future/transaction.py
Comment (algorandskiy) [inline algosdk/future/transaction.py:3027]: Please add a new error type `ConfirmationTimeout` or similar

File: algosdk/v2client/algod.py
Comment (algorandskiy) [inline algosdk/v2client/algod.py:376]: all these changes are not needed

**Review by algorandskiy — CHANGES_REQUESTED** [review]
Thank you for the submission. Please update the function interface as suggested.

File: algosdk/future/transaction.py
Comment (fergalwalsh) [inline algosdk/future/transaction.py:3033]: It would be useful to return the `tx_info` here. Often there is a need to check the some properties of this after waiting.

File: algosdk/future/transaction.py
Comment (jasonpaulos) [inline algosdk/future/transaction.py:3017]: `algod_client` should be listed here too

**Review by jasonpaulos — COMMENTED** [review]
Please add the Algod v2 client to the docstring and fix the formatting issues, and it will be ready to merge.

**Review by algorandskiy — APPROVED** [review]
LGTM. Formatting issues
```
$ black --check .
would reformat algosdk/v2client/algod.py
would reformat algosdk/future/transaction.py
would reformat test/steps/steps.py
Oh no! 💥 💔 💥
3 files would be reformatted, 61 files would be left unchanged.
The command "black --check ." exited with 1.
```

**Comment by algorandskiy** [comment]
Please merge-in develop

**Comment by DamianB-BitFlipper** [comment]
Github is telling me: "Only those with write access to this repository can merge pull requests." What should I do in this case?

**Comment by jasonpaulos** [comment]
@algorandskiy meant to merge develop into this branch so it's up to date. However, the PR can still be merged automatically by github, so no need to do this.

@DamianB-BitFlipper can you please add `algod_client` to the args section of the `wait_for_confirmation` function? After that it looks good to merge to me

**Comment by algorandskiy** [comment]
Ah, I see, I had "rebase and merge" option selected for some reason and it indicated conflicts. "squash and merge" is green - so please ignore my last comment.

**Comment by DamianB-BitFlipper** [comment]
Should be pushed with the doc comment updated. I do not have write permissions to this repo, so I cannot be the one merging (I believe).

**Comment by jasonpaulos** [comment]
Thank you for updating the docstring

**Pull request merged**
